### PR TITLE
nac 547 Upgrade .NET 6 code base and IaC.

### PR DESCRIPTION
**Pull request by kcherukupalli**
## What’s changing?

Update code base to .NET 6.x and Azure Function runtime 4.x. Updated all references to then major release for dependabot process.
Updated IAC documentation with latest .Net 6 and Function related  things.

Other documentation is part of another PR.

## Why?

Related to nac-547

## This PR has:

- [ x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [x ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

## Anything else?
We need to  `Purge` any Azure Functions running on 3 before running the IAC script.   If not it would fail with the following message: `You're trying to use v3 tooling to publish to a non-v3 function app (FUNCTIONS_EXTENSION_VERSION is set to ~4)`.

### CODE REVIEW

**Review by dborgen2 — APPROVED** [review]
It looks good to me, but probably best if someone more familiar with IAC approves as well.

File: iac/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-FROM mcr.microsoft.com/dotnet/sdk:3.1-focal AS build-env
+FROM image: mcr.microsoft.com/dotnet/sdk:6.0-focal AS build-env
 #Copy IAC scripts to prepare the deployment enviroment
 COPY ./iac/docker/* ./iac/
 COPY ./iac/install-extensions.bash ./iac/
 COPY ./tools/common.bash ./tools/
 #Set to an working Azure CLI version
 #If you want lastest version set the set the AZURE_CLI_VERSION to 0
-ENV AZURE_CLI_VERSION=2.30.0-1
```
Comment (bgfoshay) [inline iac/Dockerfile:8]: Question: Just curious, is the CLI upgrade necessary?

Comment (ryangeorgeharvey) [reply iac/Dockerfile:8]: I think using the older version might have caused Krishna to get the error with '--mi-system-assigned'. Newer versions have added support for that argument.

Comment (kcherukupalli) [reply iac/Dockerfile:8]: I used Windows version of  IaC for deployment.

Comment (bgfoshay) [reply iac/Dockerfile:8]: I got past this by updating line 27 in install-azure-cli.bash script from

    apt-get install azure-cli="$AZURE_CLI_VERSION"~focal -V
to
    apt-get install azure-cli="$AZURE_CLI_VERSION"~bullseye -V

Comment (kcherukupalli) [reply iac/Dockerfile:8]: updated the file and  pushed.

**Review by ryangeorgeharvey — COMMENTED** [review]
Issue: https://github.com/18F/piipan/blob/5f8ccb9bc9c4521d7c3e69fda4b92305eda65791/.vscode/settings.json#L4 Should this be changed to 4?

**Comment by ryangeorgeharvey** [comment]
> 

Same changed needed for etl\src\Piipan.Etl\.vscode\settings.json

**Review by ryangeorgeharvey — COMMENTED** [review]
Do you plan on handling the changes to https://github.com/18F/piipan/blob/nac-547-upgrade-net6/.github/dependabot.yml in a different PR?

**Comment by kcherukupalli** [comment]
> 

Need to add the new Component projects,  will add that along with documentation PR.  Thank you for pointing out.

**Comment by kcherukupalli** [comment]
> > 
> 
> Same changed needed for etl\src\Piipan.Etl.vscode\settings.json

Fixed in both the places.

**Review by bgfoshay — APPROVED** [review]

**Pull request merged**